src/store.ts

```diff
@@ -229,108 +236,139 @@ class Store extends ElectronStore<StoreData> {
     }
 
     /**
-     * Prepare the safeStorage backend for use.
-     * We don't eagerly import from keytar as that would bring in data for all Element profiles and not just the current one,
-     * so we import lazily in getSecret.
+     * Normalise the backend to a sane value (exclude `unknown`), respect forcePlaintext mode,
+     * and ensure that if an encrypted backend is picked that encryption is available, falling back to plaintext if not.
+     * @param forcePlaintext - whether to force plaintext mode
+     * @private
      */
-    private async prepareSafeStorage(): Promise<void> {
-        await app.whenReady();
+    private chooseBackend(forcePlaintext: boolean): SaneSafeStorageBackend {
+        if (forcePlaintext) {
+            return "plaintext";
+        }
 
-        let safeStorageBackend = this.get("safeStorageBackend");
+        const isEncryptionAvailable = safeStorage.isEncryptionAvailable();
         if (process.platform === "linux") {
+            // The following enables plain text encryption if the backend used is basic_text.
+            // It has no significance for any other backend.
+            // We do this early so that in case we end up using the basic_text backend (either because that's the only one available
+            // or as a fallback when the configured backend lacks encryption support), encryption is already turned on.
+            safeStorage.setUsePlainTextEncryption(true);
+
             // Linux safeStorage support is hellish, the support varies on the Desktop Environment used rather than the store itself.
             // https://github.com/electron/electron/issues/39789 https://github.com/microsoft/vscode/issues/185212
-            const selectedSafeStorageBackend = safeStorage.getSelectedStorageBackend();
-            console.info(
-                `safeStorage backend '${selectedSafeStorageBackend}' selected, '${safeStorageBackend}' in config.`,
-            );
-
-            if (selectedSafeStorageBackend === "unknown") {
-                // This should never happen but good to be safe
-                await dialog.showMessageBox({
-                    title: _t("store|error|unknown_backend_override_title"),
-                    message: _t("store|error|unknown_backend_override"),
-                    detail: _t("store|error|unknown_backend_override_details"),
-                    type: "error",
-                });
-                throw new Error("safeStorage backend unknown");
-            }
+            const selectedBackend = safeStorage.getSelectedStorageBackend();
 
-            if (this.get("safeStorageBackendMigrate")) {
-                return this.upgradeLinuxBackend2();
+            if (selectedBackend === "unknown" || !isEncryptionAvailable) {
+                return "plaintext";
```

Comment:
So if electron detects a proper backend (i.e != `basic_text` or `unknown`) but `isEncryptionAvailable` is false due to keyring issues in linux, we'd fallback to plaintext. We could potentially relaunch the app to explicitly use `basic_text` in this case.

Reply:
It's perfectly fine to use `plaintext` as fallback instead of  `basic_text` since the encryption done isn't really secure anyway. Just want to be sure that this is intentional.

Reply:
> We could potentially relaunch the app to explicitly use basic_text in this case.

Interesting idea, will give it a go